File: cpp/check-if-n-and-its-double-exist.cpp
/* Problem url: https://leetcode.com/problems/check-if-n-and-its-double-exist
 * Code by: ldcduc
 * */
/* Begin of Solution */
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        for (int i = 0; i < arr.size(); ++ i) {
            for (int j = 0; j < arr.size(); ++ j) {
                if (i != j && arr[i] * 2 == arr[j]) {
                    return true;
                }
            }
        }    
        return false;
    }
};
/* End of Solution */
/*
 * Comment by ldcduc
 * Suggested tags: array
 * 
 * */

File: rust/check-if-n-and-its-double-exist.rs
impl Solution {
    pub fn check_if_exist(arr: Vec<i32>) -> bool {
        for i in 0..arr.len() {
            for j in 0..arr.len() {
                if i != j && arr[i] * 2 == arr[j] {
                    return true;
                }
            }
        }    
        
        false
    }
}

